src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,175 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidDataException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+    private static User user1 = new User();
+    private static User user2 = new User();
+    private static User user3 = new User();
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+        user1.setLogin("FirstUser");
+        user1.setAge(34);
+        user1.setPassword("45jisf3");
+        user2.setLogin("SecondUser");
+        user2.setAge(18);
+        user2.setPassword("newark");
+        user3.setLogin("ThirdUser");
+        user3.setAge(28);
+        user3.setPassword("vo*edekmybr");
+        storageDao.add(user1);
+        storageDao.add(user2);
+        storageDao.add(user3);
+    }
+
+    @Test
+    void addUserWithUniqueLogin_Ok() {

```

Comment:
Follow pattern for naming:
```
For this task use such convention: <methodUnderTest>_<state>_<expectedBehavior>; For example, if we are testing the method register with a null
user's age the test method name should be register_nullAge_notOk. notOk is because the test expects the register method to throw an exception.
```